pkg/querier/blocks_store_queryable.go

```diff
@@ -676,25 +659,24 @@ func canBlockWithCompactorShardIndexContainQueryShard(queryShardIndex, queryShar
 	return true, false
 }
 
-func (q *blocksStoreQuerier) fetchSeriesFromStores(
-	ctx context.Context,
-	sp *storage.SelectHints,
-	clients map[BlocksStoreClient][]ulid.ULID,
-	minT int64,
-	maxT int64,
-	matchers []*labels.Matcher,
-	convertedMatchers []storepb.LabelMatcher,
-	maxChunksLimit int,
-	leftChunksLimit int,
-) ([]storage.SeriesSet, []ulid.ULID, storage.Warnings, int, error) {
+// fetchSeriesFromStores fetches series satisfying the given matchers (convertedMatchers) and belonging to the given time range (minT, maxT)
+// from all the given store-gateways (clients). Series are fetched from the given set of store-gateways concurrently.
+// In successful case, i.e., when all the concurrent fetches terminate with no exception, fetchSeriesFromStores returns:
+//  1. a slice of fetched storage.SeriesSet
+//  2. a slice of ulid.ULID corresponding to the queried blocks
+//  3. storage.Warnings encountered during the operation
+//
+// In case of a serious error during any of the concurrent executions, the error is returned. Errors while creating storepb.SeriesRequest,
+// context cancellation, and unprocessable requests to the store-gateways (e.g., if a chunk or series limit is hit) are
+// considered serious errors. All other errors are ignored, and give rise to fetch retrials.
```

Comment:
If they are retried is that not different to ignored?

Reply:
I think by ignored he means they're not returned. The consistency check for some blocks will fail and so querying those blocks will be retried on different store-gateways.